In [1]:
import os
import numpy as np
import pandas as pd


#get data
# from pydob.exploratory import get_permits_type_df
from pydob.exploratory import (
                    get_year_counts
            )
from pydob.database import get_query_as_df
# plotting
import matplotlib.pyplot as plt
from pydob.settings import nt_style, nt_blue, nt_black

from pydob.database import get_permits_now

In [2]:
%matplotlib inline
plt.style.use(nt_style)

# 7. DOB NOW
- we need to rope in the DoB NOW stuff for applications and permits. let's get applications and permits numbers that are the sum of the NOW and non-NOW tables. in particular, it would be great to get our permit issuance number vs rate table updated with these numbers



- [DOB NOW: Build – Job Application Filings](https://data.cityofnewyork.us/Housing-Development/DOB-NOW-Build-Job-Application-Filings/w9ak-ipjd): List of most job filings filed in DOB NOW. This dataset does not include certain types of job.
    - No dates data

- [DOB NOW: Electrical Permit Applications](https://data.cityofnewyork.us/City-Government/DOB-NOW-Electrical-Permit-Applications/dm9a-ab7w): This dataset is part of the DOB NOW Electrical Permit Data Collection

- [DOB NOW: Build – Approved Permits](https://data.cityofnewyork.us/Housing-Development/DOB-NOW-Build-Approved-Permits/rbx6-tga4): List of all approved permits in DOB NOW
    - dates data included:
        - `Approved Date`: This is the date that the entire job was approved by the Plan Examiner. The applicant can now pull a permit.
        
        - `Issued Date`: This is the date that the permit was issued.
        - `Expired Date`: This is the date that the permit expires.


In [3]:
# dob_now_applied = pd.read_csv("/Users/francescao/Downloads/DOB_NOW__Build___Job_Application_Filings.csv")

In [4]:
permits_now_year_counts = get_year_counts(index_col='job_filing_number',
                                          year_col='issued_date_year', 
                                          dataset_name='permits_now')


permits_now_year_counts.columns = ['approved_counts']


In [5]:
permits_now_year_counts

approved_counts
issued_date_year                 
2016                            8
2017                         2213
2018                        26440
2019                        40187
2020                         8081

**Note:** 
- We find that in DOB NOW approved permits, the approved date started from 2016 June. There were 8 cases approved in 2016, 2213 approved in 2017 and with a jump increase in 2018 and 2019.

- This means that more and more people starts to use DOB NOW as an approach to apply for permits. 


In [24]:
permits_now.permittees_license_type.value_counts()

GC    59223
S      7481
P      6074
F      2021
O       211
RA       59
PE       52
PR       31
Name: permittees_license_type, dtype: int64

In [25]:
permits_now.work_type.value_counts()

Sidewalk Shed         23747
Construction Fence    16350
Supported Scaffold    12168
Sign                   7484
Plumbing               5845
Mechanical Systems     3025
Antenna                2064
Sprinklers             1815
Curb Cut               1164
Structural              844
Boiler Equipment        439
Standpipe               207
Name: work_type, dtype: int64